Attaching a logGroomerSidecar to dag processor

wookiist opened the pull request:
## Description
This PR attaches the logGroomer sidecar pod when using a standalone `dag processor`. This is to prevent scheduler logs from growing infinitely in the `logs` directory of that `dag processor` pod. 

In fact, one of the Airflow clusters my team uses had about `3.5 TiB` of scheduler logs accumulated in the emptyDir of a `dag processor` pod, which reduced the ephemeral-storage availability on that node to the point of pod eviction, resulting in a pod eviction.

```
airflow@airflow-test-dag-processor-78f9bfdb88-hmckb:/opt/airflow/logs$ ls
scheduler
airflow@airflow-test-dag-processor-78f9bfdb88-hmckb:/opt/airflow/logs$ du -sh
3.5T	     .
```

We haven't figured out why the standalone `dag processor` was accumulating scheduler logs, but we think it's a good idea to attach a logGroomer sidecar like any other scheduler pod or worker pod to prevent this from happening in the first place. 

In this PR, we've modified the helm chart to attach a logGroomer to the standalone `dag processor`.

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

vincbeck committed:
Remove gauge scheduler.tasks.running (#30374)

* Remove gauge scheduler.tasks.running

* Add significant.rst file

* Update newsfragments/30374.significant.rst

---------

Co-authored-by: Niko Oliveira <onikolas@amazon.com>

jaewookoh-toss committed:
Attaching a logGroomerSidecar to Dag Processor

jaewookoh-toss committed:
Update dagProcessor enabled false

hussein-awala committed:
Recover from `too old resource version exception` by retrieving the latest `resource_version` (#30425)

* Recover from `too old resource version exception` by retrieving the latest `resource_version`

* Update airflow/executors/kubernetes_executor.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

error418 committed:
docs: use correct import path for Dataset (#30617)

MarosMK committed:
Speed up TaskGroups with caching property of group_id (#30284)

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

dylanbstorey committed:
Fix `TriggerDagRunOperator` with deferrable parameter (#30406)

* readding after borked it

* pre-commit

* finally fixing after the github issue last week

* push fix

* feedback from hussein

potiuk committed:
Fix failing SQS tests on moto upgrade (#30625)

The new moto (4.1.7) performs additional validation on the queues
created during tests and it failes the tests when content
deduplication is not specified.

Explicit setting the deduplication mode, fixes the problem and
allows the new moto to be installed.

blinkseb committed:
fix possible race condition when refreshing DAGs (#30392)

* fix possible race condition when refreshing DAGs

* merge the two queries into one

* Remove provide_session from internal function

Since get_latest_version_hash_and_updated_datetime is internal and we
always pass in the session anyway, the provide_session decorator is
redundant and only introduce possibility for developer errors.

---------

Co-authored-by: Sébastien Brochet <sebastien.brochet@nielsen.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

jedcunningham committed:
Remove Norm and Hussein from the triage group (#30627)

Hussein is now a commiter and Norm has completed building out the
initial AIP-52 tasks.

moiseenkov committed:
Remove mysql-connector-python (#30487)

* Turn the package 'mysql-connector-python' as an optional feature

* Update airflow/providers/mysql/provider.yaml

* Update airflow/providers/mysql/CHANGELOG.rst

---------

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

potiuk committed:
Better error message where non-compatible providers are not excluded (#30629)

When compatibility check is performed for old version of Airflow,
we attempt to install all providers for the old version. However if
one of the providers has >= limit on Airflow for newer version of
Airflow, this installation lead to attempting to upgrade airflow
rather than fail, which could lead to misleading errors.

This change adds "airflow==x.y.z" taken from --use-airflow-version
flag to the `pip install` command, which should in this case fail
with much more accurate message, that the provider conflicts with
airflo version.

shanecglass committed:
Updating the links to the Dataform product documentation to fix 404 redirect error (#30631)

Updating the links to the Dataform product documentation to fix 404 redirect error

jaewookoh-toss committed:
Update dagProcessor logGroomer value comments

mrichman committed:
New AWS sensor — DynamoDBValueSensor (#28338)

jlaneve committed:
Remove duplicate param docstring in EksPodOperator (#30634)

oboki committed:
In `DockerOperator`, adding an attribute `tls_verify` to choose whether to validate certificate (#30309) (#30310)

* add `tls_verify` to choose whether to validate certificate (#30309)

---------

Co-authored-by: Hussein Awala <hussein@awala.fr>

hussein-awala committed:
Add `max_active_tis_per_dagrun` for Dynamic Task Mapping (#29094)

* add max_active_tis_per_dagrun param to BaseOperator

* set has_task_concurrency_limits when max_active_tis_per_dagrun is not None

* check if max_active_tis_per_dagrun is reached in the task deps

* check if all the tasks have None max_active_tis_per_dagrun before auto schedule the dagrun

* check if the max_active_tis_per_dagrun is reached before queuing the ti

* check max_active_tis_per_dagrun in backfill job

* fix current tests and ensure everything is ok before adding new tests

* refacto TestTaskConcurrencyDep

* fix a bug in TaskConcurrencyDep

* test max_active_tis_per_dagrun in TaskConcurrencyDep

* tests max_active_tis_per_dagrun in TestTaskInstance

* test dag_file_processor with max_active_tis_per_dagrun

* test scheduling with max_active_tis_per_dagrun on different DAG runs

* test scheduling mapped task with max_active_tis_per_dagrun

* test max_active_tis_per_dagrun with backfill CLI

* add new starved_tasks filter to avoid affecting the scheduling perf

* unify the usage of TaskInstance filters and use TI

* refacto concurrecy map type and create a new dataclass

* move docstring to ConcurrencyMap class and create a method for default_factory

* move concurrency_map creation to ConcurrencyMap class

* replace default dicts by counters

* replace all default dicts by counters in the scheduler_job_runner module

* suggestions from review

RNHTTR committed:
Simplify logic to resolve tasks stuck in queued despite stalled_task_timeout (#30375)

* simplify and consolidate logic for tasks stuck in queued

* simplify and consolidate logic for tasks stuck in queued

* simplify and consolidate logic for tasks stuck in queued

* fixed tests; updated fail stuck tasks to use run_with_db_retries

* mypy; fixed tests

* fix task_adoption_timeout in celery integration test

* addressing comments

* remove useless print

* fix typo

* move failure logic to executor

* fix scheduler job test

* adjustments for new scheduler job

* appeasing static checks

* fix test for new scheduler job paradigm

* Updating docs for deprecations

* news & small changes

* news & small changes

* Update newsfragments/30375.significant.rst

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Update newsfragments/30375.significant.rst

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* added cleanup stuck task functionality to base executor

* fix sloppy mistakes & mypy

* removing self.fail from base_executor

* Update airflow/jobs/scheduler_job_runner.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Update airflow/jobs/scheduler_job_runner.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Fix job_id filter

* Don't even run query if executor doesn't support timing out queued tasks

* Add support for LocalKubernetesExecutor and CeleryKubernetesExecutor

* Add config option to control how often it runs - we want it quicker than
the timeout

* Fixup newsfragment

* mark old KE pending pod check interval as deprecated by new check interval

* Fixup deprecation warnings

This more closely mirrors how deprecations are raised for "normal"
deprecations.

I've removed the depth, as moving up the stack doesn't really help the
user at all in this situation.

* Another deprecation cleanup

* Remove db retries

* Fix test

---------

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
Co-authored-by: Jed Cunningham <jedcunningham@apache.org>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

Commit:
Display Video 360 cleanup v1 API usage (#30577)

* Display Video 360 cleanup v1 API usage

* Update docs

hussein-awala committed:
Fix mapped tasks partial arguments when DAG default args are provided (#29913)

* Add a failing test to make it pass

* use partial_kwargs when they are provide and override only None values by dag default values

* update the test and check if the values are filled in the right order

* fix overriding retry_delay with default value when it is equal to 0

* add missing default value for inlets and outlets

* set partial_kwargs dict type to dict[str, Any] and remove type ignore comments

* create a dict for default values and use NotSet instead of None to support None as accepted value

* update partial typing by removing None type from some args and set NotSet for all args

* Tweak kwarg merging slightly

This should improve iteration a bit, I think.

* Fix unit tests

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

mobuchowski committed:
First commit of OpenLineage provider. (#29940)

This PR consistent mostly of code that was created in OpenLineage project. It 
consists of

- Provider wiring
- OpenLineageListener that uses Listener API to get notification about changes
  to TaskInstance and Dag states
- Extractor framework, which is used to extract lineage information from
  particular operators. It's ment to be replaced by direct implementation of
  lineage features in later phase and extracting them using DefaultExtractor.
  This PR does not include actual extractors, but code around using and registering them.
- OpenLineageAdapter that translates extracted information to OpenLineage events.
- Utils around specific Airflow OL facets and features

This is a base implementation that's not ment to be released yet, but to add
code modified to be consistent with Airflow standards, get early feedback and
provide canvas to add later features, docs, tests on.

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

ephraimbuddy committed:
Add v2-6-test and v2-6-stable to codecov and protected branches (#30640)

amoghrajesh committed:
Adding configuration to control retry parameters for k8s api client (#29809)

* Adding configuration to control retry parameters for k8s api client

* Handling review comments

* Fixing code bug

* Fixing failing tests

* Temporary commit with UT wip

* Fixing unit test

* Fixing the strict checks

* Handling review comments from Hussein

* Revert "Handling review comments from Hussein"

This reverts commit fa3bc260f7462c42620f694ee97b7f15c0b0b9c3.

* Fixing failing ut

* Reverting bad hack

* Updating logic in kube_client.py

Co-authored-by: Hussein Awala <hussein@awala.fr>

* Fixing unit tests

* Fixing unit tests

* Handling review comments from Ash

* Fix loading mock call args for python3.7

* Apply suggestions from code review

* fix static check

* add in 2.6.0

---------

Co-authored-by: Amogh <adesai@cloudera.com>
Co-authored-by: Hussein Awala <houssein.awala.96@gmail.com>

jedcunningham reviewed (changes requested):
Can you add test coverage? Follow [this example](https://github.com/apache/airflow/blob/dcffbb4aff090e6c7b6dc96a4a68b188424ae174/tests/charts/test_scheduler.py#L656-L658).

mikaeld committed:
fix(chart): webserver probes timeout and period. (#30609)

* fix(chart): webserver probes timeout and period

* Update default values in JSON schema to reflect values.yaml

* remove defautl templated values

eladkal committed:
Clarify release announcements on social media (#30639)

utkarsharma2 committed:
DynamoDBHook - waiter_path() to consider `resource_type` or `client_type` (#30595)

* Add  while initilizing

* Add  while initilizing

* Add logic to pick either client_type or resource_type

* Add test case

* Assert expected path

bbovenzi committed:
Improve task & run actions ux in grid view (#30373)

* update run clear+mark, update task clear

* add mark as tasks and include list of affected tasks

* Add support for mapped tasks, add shared modal component

* Clean up styling, restore warning for past/future tg clear

maahir22 committed:
Add command to get DAG Details via CLI (#30432)



---------

Co-authored-by: Hussein Awala <hussein@awala.fr>
Co-authored-by: Hussein Awala <houssein.awala.96@gmail.com>

sean-rose committed:
When clearing task instances try to get associated DAGs from database (#29065)

* When clearing task instances try to get associated DAGs from database.

This fixes problems when recursively clearing task instances across multiple DAGs:
  * Task instances in downstream DAGs weren't having their `max_tries` property incremented, which could cause downstream external task sensors in reschedule mode to instantly time out (issue #29049).
  * Task instances in downstream DAGs could have some of their properties overridden by an unrelated task in the upstream DAG if they had the same task ID.

* Use session fixture for new `test_clear_task_instances_without_dag_param` test.

* Use session fixture for new `test_clear_task_instances_in_multiple_dags` test.

---------

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

eladkal committed:
Organize Amazon providers docs index (#30541)

vandonr-amz committed:
preload airflow imports before dag parsing to save time (#30495)



---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

phanikumv committed:
Add delete inactive run functionality to databricks provider (#30646)

rohan472000 committed:
Create audit_logs.rst (#30405)



---------

Co-authored-by: Josh Fell <48934154+josh-fell@users.noreply.github.com>

pierrejeambrun committed:
Present affected task instances as table (#30633)

jedcunningham committed:
Helm chart 1.9.0 has been released (#30649)

ephraimbuddy committed:
Add 2.6.0b1 to issue template (#30652)

ying-w committed:
add missing project_id in BigQueryGetDataOperator (#30651)

jaewookoh-toss committed:
Adding loggroomer test code to test-dag-processor.py

potiuk commented on tests/charts/test_dag_processor.py:
```diff
@@ -549,3 +550,8 @@ def test_should_add_component_specific_annotations(self):
+
+class TestDagProcessorLogGroomer(LogGroomerTestBase):
+    obj_name = "dag-processor"
+    folder = "dag-processor"
```
This will likely fail static checks

wookiist replied:
Thanks for the check!
Added newline at the end of the code.

potiuk committed:
Properly classify google_vendor package to google provider (#30659)

We've recently added google_venor package to vendor-in ads library,
and we had to do it outside of regular google provider package,
because internally the library assumed our google package is top
level package when discovering the right relative imports (#30544).

This confused the pre-commit that updates provider depedencies to
not recognise the package and print warnings about bad classification.

Special case handling will classify it to google provider.

potiuk committed:
Make pandas optional in workday calendar example (#30660)

The workday calendar expected pandas to be available and it is part
of our examples, however Airflow does not have pandas as a core
dependency, so in case someone does not have pandas installed, importing
of the workday example would fail.

This change makes pandas optional and fallbacks to regular working
days for the example in case it is not available (including warning
about it). It also fixes a slight inefficiency where the
USFederalHoliday calendar has been created every time next workday
was calculated.

VladaZakharova committed:
Update Google Campaign Manager360 operators to use API v4 (#30598)

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

hussein-awala committed:
Skip KubernetesPodOperator task when it returns a provided exit code (#29000)

* Skip KubernetesPodOperator task when it returns a provided exit code

* set default value to None, and get exit code only when skip_exit_code is not None

* get the exit code for the base container and check if everything is ok

* add unit test for the operator

* add a test for deffered mode

* apply change requests

---------

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

notatallshaw committed:
Upgrade Pip to 23.1 (#30663)

potiuk committed:
Fix docs building for workday example. (#30664)

The #30660 was merged to quickly as it results in doc building
failure. This PR fixes it.

Bowrna committed:
docker compose doc changes (#30662)

potiuk committed:
Add suspended providers to pytest collection test (#30668)

Pytest collection has been extracted recently to a separate job
and SUSPENDED_PROVIDERS_FOLDERS variable was not set in the new
job - which causes suspended provider tests being attempted by
pytest collection, leading to import errors when suspended providers
have some dependencies removed from our image.

potiuk committed:
Workaround type-incompatibility with new attrs in openlineage (#30674)

The new attrs released today (11 hours ago) had added typing
information and they caused OpenLineageRedactor to fail mypy checks.

Temporary adding type: ignore should allow to upgrade to the new
attrs and stop PRs changing dependencies from failing.

Related: #30673

jaewookoh-toss committed:
Adding line at the end of the code

ephraimbuddy committed:
Update the release note (#30680)

* Update the release note

During the beta release, I observed some minor things that need fixing.
Here's the PR

* Use local import

uranusjr committed:
Correctly pass a type to attrs.has() (#30677)

phanikumv committed:
Merge WasbBlobAsyncSensor to WasbBlobSensor (#30488)

vchiapaikeo committed:
Updated app to support configuring the caching hash method for FIPS v2 (#30675)

potiuk committed:
Install twine with --force for package verification (#30683)

In some cases when the machine has been reused across builds, pipx
installed twine might seem both installed and removed (this happens
when builds are cancelled while installing twine.

Installing twine with --force should fix the problem.

burnash committed:
Fix docs: add an "apache" prefix to pip install (#30681)

vincbeck committed:
Remove unittests.TestCase from tests/test_utils (#30685)

potiuk committed:
Introduce consistency of package sequence for "Other" test type (#30682)

When packages for "Other" test type are calculated, the list of
all test folders is generated and they are compared with the
packages previously selected by the "predefined" test types. This
is done via `find` method that returns the folders in arbitrary
order, mostly depending on the sequence the folders were created.

In case the tests from some packages have some side-effects that
impact tests in other packages (obviously not something that is
desired), this might end up that the tests succeed in one
environment, but fail in another. This happened for example
in case of #30362 that had cross-package side-effect later
fixed in #30588. There - results of "Other" test type depended
on where the tests were executed.

This PR sorts the find output so it is always in consistent order.
we are using ASCII for package names and the test types are
derived in the same Docker CI image with the same LOCALE, so it
should guarantee that the output of packages for "Other" test type
should be always consistent.

vchiapaikeo committed:
Add missing version val to caching_hash_method config (#30688)

potiuk committed:
Upgrade to MyPy 1.2.0 (#30687)

Upgrading to latest (released a week ago) MyPy in the hopes it
will fix some more problem with attrs after upgrading new packages,
but it seems that even the latest MyPy does not know about the
new typing changes introduced in attrs (traditionally mypy has
attrs plugin that injects appropriate typing but apparently it
needs to catch up with those changes.

potiuk committed:
Parallelize Helm tests with multiple job runners (#30672)

Helm Unit tests are using template rendering and the rendering
uses a lot of CPU for `helm template command`. We have a lot of
those rendering tests (>800) so even running the tests in parallel
on a multi-cpu machine does not lead to a decreased elapsed time
to execute the tests.

However, each of the tests is run entirely independently and we
should be able to achieve much faster elapsed time if we run
a subset of tetsts on separate, multi-CPU machine. This will not
lower the job build time, however it might speed up elapsed time
and thus get a faster feedback.

This PR achieves that.

hussein-awala committed:
Skip PythonVirtualenvOperator task when it returns a provided exit code (#30690)

* Add a new argument to rais skip exception when the python callable exit with the same value

* add unit tests for skip_exit_code

ferruzzi committed:
[OTel Integration] Add tagging to existing stats (#30496)

hussein-awala committed:
rename skip_exit_code to skip_on_exit_code and allow providing multiple codes (#30692)

* rename skip_exit_code to skip_on_exit_code and allow providing multiple codes

* replace list type by Container

pierrejeambrun committed:
Fix d3 dependencies (#30702)

vincbeck committed:
Update system test example_emr to have logs (#30715)

mrichman committed:
Fixed logging issue (#30703)

Co-authored-by: Mark Richman <mrkrchm@amazon.com>

potiuk committed:
Separate out and clarify policies for providers (#30657)

This change separates out the policies we have for providers to
a separate PROVIERS.rst file. It also documents clearly the process
and policy we have for accepting new community-managed providers,
explaining the conditions that have to be fulfilled and stating
a very strong preference of keeping providers maintained by the
3rd-party providers when there are 3rd-party teams that manage
the providers.

utkarsharma2 committed:
SqlToS3Operator - Add feature to partition SQL table (#30460)

jaewookoh-toss committed:
locating extraContainers at the last
